frontend/device/android/device.lua

```diff
@@ -362,13 +362,41 @@ function Device:canExecuteScript(file)
     end
 end
 
+--swallow all events
+local function processEvents()
+    local events = ffi.new("int[1]")
+    local source = ffi.new("struct android_poll_source*[1]")
+    local poll_state = android.lib.ALooper_pollAll(-1, nil, events, ffi.cast("void**", source))
+    if poll_state >= 0 then
+        if source[0] ~= nil then
+            if source[0].id == C.LOOPER_ID_MAIN then
+                local cmd = C.android_app_read_cmd(android.app)
+                C.android_app_pre_exec_cmd(android.app, cmd)
+                C.android_app_post_exec_cmd(android.app, cmd)
+            elseif source[0].id == C.LOOPER_ID_INPUT then
+                local event = ffi.new("AInputEvent*[1]")
+                while android.lib.AInputQueue_getEvent(android.app.inputQueue, event) >= 0 do
+                    if android.lib.AInputQueue_preDispatchEvent(android.app.inputQueue, event[0]) == 0 then
+                        android.lib.AInputQueue_finishEvent(android.app.inputQueue, event[0], 1)
+                    end
+                end
+            end
+        end
+    end
+end
+
 function Device:showLightDialog()
     local usleep = require("ffi/util").usleep
     local title = android.isEink() and _("Frontlight settings") or _("Light settings")
     android.lights.showDialog(title, _("Brightness"), _("Warmth"), _("OK"), _("Cancel"))
     repeat
+        processEvents() -- swallow all events, including the last one
```

Comment:
in a loop? or just once before `android.lights.showDialog` ?

Reply:
>in a loop?

I am afraid yes.
Doing the processEvents() before or after showDialog and doing the processEvents() after a usleep(50000) **not** in the loop does not help. Instead I am getting different events (like single taps).
If you are concerned about the load we could lengthen the usleep to 75000.

Reply:
> If you are concerned about the load we could lengthen the usleep to 75000.

Nope. I'm trying to figure out what happens. Could you attach a logcat of a debug build (it needs to be debug so it logs changes in activity state).

IIRC the dialog stays on top an catches all input events while it is running, so they're not forwarded to native code. Maybe that's not true anymore with the native surface implementation. Anyways using https://github.com/koreader/koreader/blob/master/frontend/device/android/device.lua#L271-L273 should get rid of noisy input events and shouldn't affect dialog controls.

Reply:
@pazos: The `setIgnoreInput` was the first trick I tired, before all the event things. Nope, did not change anything.

Sorry, the compilation of the debug apk, took long. (It's time to parallelize the make.)

Reply:
setIgnoreInput won't help to empty the event pool, but can prevent to fill it again with new events.

> These are the logcats. logcat -c; logcat > logcat1.txt; start koreader -> long press on edge -> open of the dialog -> change brightness, warmth -> close dialog. Did it two times.

Thanks for the logs!. As they don't show any crash I think you made them with the workaround. I'm sadly unable to reproduce the original issue on the emulator:

![test](https://user-images.githubusercontent.com/975883/91606450-5f249d00-e972-11ea-813a-b0a1797d5ea6.gif)

(FWIW I'm holding the down right corner to show the dialog)